ee/localserver/request-query.go

```diff
@@ -19,48 +21,100 @@ func (ls *localServer) requestQueryHanlderFunc(w http.ResponseWriter, r *http.Re
 		return
 	}
 
-	var body map[string]any
+	var body map[string]string
 	if err := json.NewDecoder(r.Body).Decode(&body); err != nil {
 		sendClientError(w, fmt.Sprintf("error unmarshaling request body: %s", err))
 		return
 	}
 
-	queryRaw, ok := body["query"]
-	if !ok {
+	query, ok := body["query"]
+	if !ok || query == "" {
 		sendClientError(w, "no query key found in request body json")
 		return
 	}
 
-	query, ok := queryRaw.(string)
-	if !ok {
-		sendClientError(w, fmt.Sprintf("query value not a string: %s", query))
+	var results []map[string]string
+	var err error
+	backoff.WaitFor(func() error {
+		results, err = ls.querier.Query(query)
+		return err
+	}, 1*time.Second, 250*time.Millisecond)
+
+	if err != nil {
+		sendClientError(w, fmt.Sprintf("error executing query: %s", err))
 		return
 	}
 
-	if query == "" {
-		sendClientError(w, "empty query")
+	jsonBytes, err := json.Marshal(results)
+	if err != nil {
+		sendClientError(w, fmt.Sprintf("error marshalling results to json: %s", err))
 		return
 	}
 
+	w.Write(jsonBytes)
+}
+func (ls *localServer) requestScheduledQueryHandler() http.Handler {
+	return http.HandlerFunc(ls.requestScheduledQueryHandlerFunc)
+}
+
+// requestScheduledQueryHandlerFunc uses the name field in the request body to look up
+// an existing osquery scheduled query execute it, returning the results.
+func (ls *localServer) requestScheduledQueryHandlerFunc(w http.ResponseWriter, r *http.Request) {
+	// The driver behind this is that the JS bridge has to use GET requests passing the query (in a nacl box) as a URL parameter.
+	// This means there is a limit on the size of the query. This endpoint is intended to be a work around for that. It ought to work like this:
+	//
+	// 1. K2 looks up the name of a osquery scheduled query it want to be run. The same query should be available to launcher in the osquery_schedule table.
+	// 2. K2 calls `/scheduledquery`, with body `{ "name": "name_of_scheduled_query" }`
+	// 3. Launcher looks up the query's sql using the provided name
+	// 4. Launcher executes the query
+	// 5. Launcher returns results
+
+	if r.Body == nil {
+		sendClientError(w, "request body is nil")
+		return
+	}
+
+	var body map[string]string
+	if err := json.NewDecoder(r.Body).Decode(&body); err != nil {
+		sendClientError(w, fmt.Sprintf("error unmarshaling request body: %s", err))
+		return
+	}
+
+	name, ok := body["name"]
+	if !ok || name == "" {
+		sendClientError(w, "no name key found in request body json")
+		return
+	}
+
+	scheduledQueryQuery := fmt.Sprintf("select query from osquery_schedule where name = '%s'", name)
```

Comment:
I think this would be interesting as a `LIKE` so we can pull more than one set of results.

Reply:
@directionless, idea (possibly bad) 💡 

what if we just let accepted a query for the query? something like:

`/fetchedquery --data '{"query": "select query from osquery_scheduled where name like '%foo%"}'`

we would have to handle the case where the query in the body returns more than one column ... error or first alphabetically or something

Reply:
I'm not sure it buys us much. 

The query names are, in theory, known. But there's some skew that comes from how everything is generated. But a `LIKE` should cover pretty much anything. 

Go really far down the dynamic query and it just get confusing. Would we support other tables? Full sql? Anything that returns rows with a `query` column? 

I guess I could see some nebulous future where sometimes you wanted `select query from osquery_scheduled where name like '%foo%"`, but sometimes you wanted a pivot on `kolide_server_data`.

But mostly I think it's complicated